src/EditorFeatures/CSharp/AutomaticCompletion/AutomaticLineEnderCommandHandler.cs

```diff
@@ -41,6 +41,40 @@ protected override void NextAction(IEditorOperations editorOperation, Action nex
             editorOperation.InsertNewLine();
         }
 
+        protected override bool TreatAsReturn(Document document, int position, CancellationToken cancellationToken)
+        {
+            var root = document.GetSyntaxRootAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+
+            var endToken = root.FindToken(position);
+            if (endToken.IsMissing)
+            {
+                return false;
+            }
+
+            var tokenToLeft = root.FindTokenOnLeftOfPosition(position);
+            var startToken = endToken.GetPreviousToken();
+
+            // case 1:
```

Comment:
Not any more that I know of. Just structured the code to be so that eventually it can `return case1 || case 2 .. `